sonarts-core/docs/rules/no-unconditional-jump.md

```diff
@@ -1,32 +1,39 @@
 # no-unconditional-jump
 
-Having an unconditional `break`, `return` or `throw` in a loop renders it useless; the loop will only execute once and the loop structure itself is simply wasted keystrokes.
-Having an unconditional `continue` in a loop is itself wasted keystrokes.
-For these reasons, unconditional jump statements should never be used except for the final `return` in a function or method.
+A loop with at most one iteration is equivalent to the use of an `if` statement to conditionally execute one piece of code. No developer expects to find such a use of a loop statement. If the initial intention of the author was really to conditionally execute one piece of code, an `if` statement should be used instead.
+
+At worst that was not the initial intention of the author and so the body of the loop should be fixed to use the nested `return`, `break` or `throw` statements in a more appropriate way.
 
 ## Noncompliant Code Example
 
 ```typescript
-for (i = 0; i < 10; i++) {
+for (int i = 0; i < 10; i++) { // noncompliant, loop only executes once
```

Comment:
why adding type in the loop? should be just `let` perhaps?

Reply:
@saberduck it's java leftover...)